File: frontend/src/lib/synthesis.ts
import { cardinalPhraseFr, formatDistance } from './geo'
import * as georisques from './georisques'
import { surveyNearbyParcels, type ParcelSurvey } from './parcelles'
import { levelFromArgiles, levelFromCount, levelFromFloodSignals, levelFromRadon, levelFromSsp, levelFromZonageSismique, worstLevel } from './rules'
import type { AddressResult, SensitivityLevel, SensitivityReport, ThemeItem, ThemeSynthesis } from '../types/sensitivity'

function describeLocalisation(loc: georisques.Localisation | null): string | null {
  return loc ? `à ${formatDistance(loc.distanceM)} ${cardinalPhraseFr(loc.direction)} du site` : null
}

const AVERTISSEMENT =
  "Cette synthèse s'appuie sur des données publiques (BRGM/Géorisques, IGN) " +
  "recensées à proximité de l'adresse indiquée. Elle donne une première lecture " +
  "des enjeux et ne remplace pas une étude réglementaire (étude de sols, avis " +
  'hydrogéologique, diagnostic ICPE...).'

// Cap on how many individual items (ICPE, CASIAS...) are listed per theme —
// enough to be useful, not enough to turn the page into a raw data dump.
const MAX_LISTED = 6

export async function buildSensitivityReport(address: AddressResult, rayonMetres: number): Promise<SensitivityReport> {
  const { lat, lon, citycode } = address

  const [icpeResult, sspResult, timCount, mvtResult, cavitesResult, inAziResult, catnatItems, zoneSismique, argilesExpo, radonClasseResult, parcelSurvey] =
    await Promise.all([
      georisques.fetchIcpe(lat, lon, rayonMetres),
      georisques.fetchSsp(lat, lon, rayonMetres),
      georisques.countTim(lat, lon, rayonMetres),
      georisques.fetchMvt(lat, lon, rayonMetres),
      georisques.fetchCavites(lat, lon, rayonMetres),
      georisques.inAzi(lat, lon, rayonMetres),
      citycode ? georisques.fetchCatnatInondation(citycode) : Promise.resolve(null),
      citycode ? georisques.zonageSismique(citycode) : Promise.resolve(null),
      citycode ? georisques.argilesExposition(citycode) : Promise.resolve(null),
      citycode ? georisques.radonClasse(citycode) : Promise.resolve(null),
      surveyNearbyParcels(lat, lon),
    ])

  const themes: ThemeSynthesis[] = [
    themeRisquesNaturels(inAziResult, catnatItems, mvtResult, cavitesResult, zoneSismique, argilesExpo, radonClasseResult, parcelSurvey, citycode),
    themeRisquesIndustriels(sspResult, icpeResult, timCount, citycode),
  ]

  return {
    address,
    rayon_metres: rayonMetres,
    genere_le: new Date().toISOString(),
    themes,
    niveau_global: worstLevel(themes.map((t) => t.niveau)),
    avertissement: AVERTISSEMENT,
  }
}

function formatDateFr(value: string | null): string | null {
  if (!value) return null
  const date = georisques.parseFrenchDate(value)
  return date ? date.toLocaleDateString('fr-FR') : value
}

const ZONAGE_SISMIQUE_LABELS: Record<number, string> = {
  1: 'très faible',
  2: 'faible',
  3: 'modérée',
  4: 'moyenne',
  5: 'forte',
}

const ARGILES_DESCRIPTIONS: Record<string, string> = {
  faible: "un phénomène de retrait-gonflement possible mais peu probable, sans mesure constructive particulière requise pour les bâtiments courants",
  moyen:
    'une probabilité de survenance du phénomène significative — des dispositions constructives (fondations, joints de rupture...) sont en général recommandées pour un projet neuf',
  moyenne:
    'une probabilité de survenance du phénomène significative — des dispositions constructives (fondations, joints de rupture...) sont en général recommandées pour un projet neuf',
  fort: 'une probabilité de survenance du phénomène élevée — une étude géotechnique préalable est fortement recommandée pour tout projet de construction',
  forte: 'une probabilité de survenance du phénomène élevée — une étude géotechnique préalable est fortement recommandée pour tout projet de construction',
}

const RADON_DESCRIPTIONS: Record<number, string> = {
  1: 'un potentiel faible : les teneurs en uranium des sous-sols sont basses',
  2: 'un potentiel faible à moyen, avec des facteurs géologiques pouvant faciliter le transfert du radon vers le bâti',
  3: 'un potentiel significatif : des mesures de prévention (ventilation, étanchéité des points d’entrée) sont recommandées pour le bâti',
}

/** "Risques naturels" — tout ce qui menace le site du fait du milieu
 * environnant : inondation/coulée de boue, mouvements de terrain, cavités,
 * sismicité, argiles, radon, et l'usage de produits phytosanitaires sur les
 * parcelles agricoles voisines. */
function themeRisquesNaturels(
  inAziValue: boolean | null,
  catnatItems: georisques.CatnatItem[] | null,
  mvt: georisques.ListResult<georisques.MvtItem> | null,
  cavites: georisques.ListResult<georisques.CaviteItem> | null,
  zoneSismique: number | null,
  argilesExpo: string | null,
  radonClasseValue: number | null,
  parcelSurvey: ParcelSurvey | null,
  citycode: string,
): ThemeSynthesis {
  const catnatCount = catnatItems?.length ?? null
  const niveaux: SensitivityLevel[] = [
    levelFromFloodSignals(inAziValue, catnatCount),
    levelFromCount(mvt?.total ?? null, 2, 6),
    levelFromCount(cavites?.total ?? null, 2, 6),
    levelFromZonageSismique(zoneSismique),
    levelFromArgiles(argilesExpo),
    levelFromRadon(radonClasseValue),
  ]
  const niveau = worstLevel(niveaux)
  const items: ThemeItem[] = []
  const manquantes: string[] = []

  if (inAziValue === null) {
    manquantes.push('zones inondables (AZI)')
  } else {
    items.push({
      label: 'Zone inondable recensée',
      detail: inAziValue ? 'Le secteur recoupe une zone inondable connue' : 'Aucune zone inondable recensée à proximité',
      source: 'Atlas des zones inondables — Géorisques',
    })
  }

  if (catnatItems === null) {
    manquantes.push('arrêtés catastrophe naturelle (inondation)')
  } else if (catnatItems.length === 0) {
    items.push({ label: 'Historique catastrophe naturelle — inondation', detail: 'Aucun arrêté recensé pour la commune', source: 'GASPAR — Géorisques' })
  } else {
    const shown = catnatItems.slice(0, MAX_LISTED)
    shown.forEach((arrete) => {
      const datePublication = arrete.datePublicationJo ?? arrete.datePublicationArrete
      const dates = [
        arrete.dateDebut && arrete.dateFin ? `évènement du ${formatDateFr(arrete.dateDebut)} au ${formatDateFr(arrete.dateFin)}` : null,
        datePublication ? `publié au Journal officiel le ${formatDateFr(datePublication)}` : null,
        arrete.codeNational ? `réf. ${arrete.codeNational}` : null,
      ].filter(Boolean)
      items.push({
        label: arrete.libelle,
        detail: dates.join(' — ') || 'Arrêté de catastrophe naturelle',
        source: 'GASPAR — Géorisques',
      })
    })
    if (catnatItems.length > shown.length) {
      items.push({ label: `+ ${catnatItems.length - shown.length} autre(s) arrêté(s)`, detail: '', source: 'GASPAR — Géorisques' })
    }
  }

  if (mvt === null) {
    manquantes.push('mouvements de terrain')
  } else if (mvt.items.length === 0) {
    items.push({ label: 'Mouvements de terrain recensés', detail: 'Aucun évènement recensé à proximité', source: 'BRGM/Géorisques' })
  } else {
    const shown = mvt.items.slice(0, MAX_LISTED)
    shown.forEach((m) =>
      items.push({
        label: m.type,
        detail: [m.lieu, m.dateDebut ? `daté du ${formatDateFr(m.dateDebut)}` : null, describeLocalisation(m.localisation)].filter(Boolean).join(' — '),
        source: 'BRGM/Géorisques',
      }),
    )
    if (mvt.total > shown.length) {
      items.push({ label: `+ ${mvt.total - shown.length} autre(s) évènement(s)`, detail: '', source: 'BRGM/Géorisques' })
    }
  }

  if (cavites === null) {
    manquantes.push('cavités souterraines')
  } else if (cavites.items.length === 0) {
    items.push({ label: 'Cavités souterraines recensées', detail: 'Aucune cavité recensée à proximité', source: 'BRGM/Géorisques' })
  } else {
    const shown = cavites.items.slice(0, MAX_LISTED)
    shown.forEach((c) =>
      items.push({
        label: c.nom ? `${c.type} — ${c.nom}` : c.type,
        detail: describeLocalisation(c.localisation) ?? '',
        source: 'BRGM/Géorisques',
      }),
    )
    if (cavites.total > shown.length) {
      items.push({ label: `+ ${cavites.total - shown.length} autre(s) cavité(s)`, detail: '', source: 'BRGM/Géorisques' })
    }
  }

  if (zoneSismique === null) {
    manquantes.push('zonage sismique')
  } else {
    items.push({
      label: 'Zonage sismique réglementaire',
      detail: `Zone ${zoneSismique} sur l'échelle réglementaire (1 très faible à 5 fort) — sismicité ${ZONAGE_SISMIQUE_LABELS[zoneSismique] ?? 'non classée'}`,
      source: 'Géorisques',
    })
  }

  if (!argilesExpo) {
    manquantes.push('retrait-gonflement des argiles')
  } else {
    items.push({
      label: 'Retrait-gonflement des argiles',
      detail: `Exposition ${argilesExpo.toLowerCase()} : ${ARGILES_DESCRIPTIONS[argilesExpo.trim().toLowerCase()] ?? 'à préciser selon la carte réglementaire'}`,
      source: 'Géorisques',
    })
  }

  if (radonClasseValue === null) {
    manquantes.push('potentiel radon')
  } else {
    items.push({
      label: 'Potentiel radon',
      detail: `Classe ${radonClasseValue} sur l'échelle réglementaire (1 à 3) : ${RADON_DESCRIPTIONS[radonClasseValue] ?? 'à préciser selon la carte réglementaire'}`,
      source: 'Géorisques',
    })
  }

  // Usage de produits phytosanitaires sur les parcelles agricoles voisines —
  // heuristique par défaut (pas de vérification "bio" automatisée possible,
  // voir parcelles.ts) : une parcelle qui n'est pas de la prairie/estive
  // (proxy pour l'élevage) est considérée comme probablement traitée.
  if (parcelSurvey === null) {
    manquantes.push('usage de produits phytosanitaires (RPG)')
  } else if (parcelSurvey.nearestTreated) {
    const p = parcelSurvey.nearestTreated
    const localisation = p.inside
      ? 'le site est situé au sein-même de cette parcelle'
      : `à ${formatDistance(p.distanceM)} ${p.direction ? cardinalPhraseFr(p.direction) : ''} du site`
    items.push({
      label: 'Parcelle agricole probablement traitée la plus proche',
      detail: `Culture : ${p.cropLabel} (code ${p.codeCultu}) — ${localisation}. Ni prairie/estive (élevage) ni certifiée bio à notre connaissance — statut biologique non vérifiable automatiquement.`,
      source: 'RPG — IGN/ASP',
    })
  } else if (parcelSurvey.anyParcelFound) {
    items.push({
      label: "Parcelles agricoles voisines",
      detail: "Seules des prairies/estives (probable élevage) sont recensées à proximité — pas de parcelle cultivée identifiée dans le rayon consulté.",
      source: 'RPG — IGN/ASP',
    })
  } else {
    items.push({ label: 'Parcelles agricoles voisines', detail: 'Aucune parcelle agricole (RPG) recensée à proximité', source: 'RPG — IGN/ASP' })
  }

  if (citycode) {
    items.push({
      label: 'Consulter tous les risques de la commune',
      detail: 'Portail Géorisques (arrêtés, PPR, sismicité, radon...)',
      source: 'Géorisques',
      href: georisques.communeRiskPortalUrl(citycode),
    })
  }

  let resume: string
  if (niveau === 'indeterminee') resume = "Les indicateurs de risques naturels n'ont pas pu être interrogés."
  else if (niveau === 'elevee')
    resume =
      'Un ou plusieurs indicateurs de risques naturels (inondation, mouvements de terrain, cavités, sismicité, argiles ou radon) atteignent un niveau élevé sur le secteur.'
  else if (niveau === 'moderee')
    resume = 'Le secteur présente un ou plusieurs indicateurs de risques naturels à surveiller, sans signal alarmant à ce stade.'
  else resume = 'Les indicateurs de risques naturels consultés sont globalement favorables sur le secteur.'

  return { key: 'risques_naturels', titre: 'Risques naturels', niveau, resume, items, donnees_manquantes: manquantes }
}

/** "Risques industriels" — anciennes et actuelles activités industrielles
 * ou de service susceptibles d'affecter le site : sites et sols pollués
 * (CASIAS/SIS), installations classées (ICPE), canalisations de matières
 * dangereuses. */
function themeRisquesIndustriels(
  ssp: georisques.SspResult | null,
  icpe: georisques.ListResult<georisques.IcpeItem> | null,
  timCount: number | null,
  citycode: string,
): ThemeSynthesis {
  const casiasTotal = ssp?.casias.total ?? null
  const sisTotal = ssp?.sis.total ?? null
  const icpeTotal = icpe?.total ?? null
  const niveau = worstLevel([
    ssp === null ? 'indeterminee' : levelFromCount(casiasTotal, 3, 10),
    ssp === null ? 'indeterminee' : levelFromSsp(sisTotal),
    levelFromCount(icpeTotal, 3, 10),
    levelFromCount(timCount, 2, 5),
  ])
  const items: ThemeItem[] = []
  const manquantes: string[] = []

  if (ssp === null) {
    manquantes.push('anciens sites industriels et sols pollués (BASIAS/SIS)')
  } else {
    const casiasShown = ssp.casias.items.slice(0, MAX_LISTED)
    if (casiasShown.length === 0) {
      items.push({ label: 'Anciens sites industriels', detail: 'Aucun site recensé à proximité', source: 'BASIAS/BASOL (CASIAS) — BRGM/Géorisques' })
    } else {
      casiasShown.forEach((site) =>
        items.push({
          label: site.identifiant ? `${site.nom} (${site.identifiant})` : site.nom,
          detail:
            [site.activite, site.commune, site.statut, describeLocalisation(site.localisation)].filter(Boolean).join(' — ') ||
            'Ancien site industriel ou de service',
          source: 'BASIAS/BASOL (CASIAS) — BRGM/Géorisques',
          href: site.ficheUrl ?? undefined,
        }),
      )
      if (ssp.casias.total > casiasShown.length) {
        items.push({
          label: `+ ${ssp.casias.total - casiasShown.length} autre(s) site(s) recensé(s)`,
          detail: 'Liste complète sur Géorisques',
          source: 'BASIAS/BASOL (CASIAS) — BRGM/Géorisques',
        })
      }
    }

    const sisShown = ssp.sis.items.slice(0, MAX_LISTED)
    if (sisShown.length === 0) {
      items.push({ label: "Secteurs d'information sur les sols (SIS)", detail: 'Aucun secteur recensé à proximité', source: 'SIS — Géorisques' })
    } else {
      sisShown.forEach((site) =>
        items.push({
          label: site.identifiant ? `${site.nom} (${site.identifiant})` : site.nom,
          detail: [site.commune, site.superficieM2 ? `${Math.round(site.superficieM2)} m²` : null, describeLocalisation(site.localisation)]
            .filter(Boolean)
            .join(' — '),
          source: 'SIS — Géorisques',
          href: site.ficheUrl ?? undefined,
        }),
      )
      if (ssp.sis.total > sisShown.length) {
        items.push({
          label: `+ ${ssp.sis.total - sisShown.length} autre(s) secteur(s) recensé(s)`,
          detail: 'Liste complète sur Géorisques',
          source: 'SIS — Géorisques',
        })
      }
    }
  }

  if (icpe === null) {
    manquantes.push('installations classées (ICPE)')
  } else {
    const shown = icpe.items.slice(0, MAX_LISTED)
    if (shown.length === 0) {
      items.push({
        label: "Installations classées pour la protection de l'environnement",
        detail: 'Aucune installation recensée à proximité',
        source: 'ICPE — Géorisques',
      })
    } else {
      shown.forEach((installation) =>
        items.push({
          label: installation.nom,
          detail: [
            installation.regime,
            installation.commune,
            installation.seveso,
            installation.codeNaf ? `NAF ${installation.codeNaf}` : null,
            describeLocalisation(installation.localisation),
          ]
            .filter(Boolean)
            .join(' — '),
          source: 'ICPE — Géorisques',
          href: installation.ficheUrl ?? undefined,
        }),
      )
      if (icpe.total > shown.length) {
        items.push({ label: `+ ${icpe.total - shown.length} autre(s) installation(s)`, detail: 'Liste complète sur Géorisques', source: 'ICPE — Géorisques' })
      }
    }
  }

  if (timCount === null) {
    manquantes.push('canalisations de transport de matières dangereuses')
  } else {
    items.push({
      label: 'Transport de matières dangereuses par canalisation',
      detail: timCount ? `${timCount} canalisation(s) recensée(s) à proximité` : 'Aucune canalisation recensée à proximité',
      source: 'TIM — Géorisques',
    })
  }

  if (citycode) {
    items.push({
      label: 'Consulter le rapport de risques complet de la commune',
      detail: 'Portail Géorisques (installations classées, sites pollués, canalisations...)',
      source: 'Géorisques',
      href: georisques.communeRiskPortalUrl(citycode),
    })
  }

  let resume: string
  if (niveau === 'indeterminee') resume = "Les indicateurs d'activités industrielles n'ont pas pu être interrogés."
  else if (niveau === 'elevee')
    resume =
      "Un ou plusieurs signaux forts d'activité industrielle sont recensés à proximité immédiate (secteur d'information sur les sols, installations classées ou canalisations en nombre)."
  else if (niveau === 'moderee')
    resume = "Un ou plusieurs signaux d'activité industrielle passée ou présente sont recensés à proximité, sans concentration particulière."
  else resume = "Aucun signal notable d'activité industrielle passée ou présente n'est recensé à proximité dans les bases publiques."

  return { key: 'risques_industriels', titre: 'Risques industriels', niveau, resume, items, donnees_manquantes: manquantes }
}

File: backend/app/synthesis.py
import asyncio

import httpx

from app.providers.georisques import GeorisquesClient
from app.rules import (
    level_from_argiles,
    level_from_count,
    level_from_flood_signals,
    level_from_radon,
    level_from_ssp,
    level_from_zonage_sismique,
)
from app.schemas import AddressResult, SensitivityReport, ThemeItem, ThemeSynthesis, worst_level


async def build_sensitivity_report(
    client: httpx.AsyncClient, address: AddressResult, rayon_metres: int
) -> SensitivityReport:
    georisques = GeorisquesClient(client)
    lat, lon, code_insee = address.lat, address.lon, address.citycode

    (
        basias_count,
        ssp_count,
        icpe_count,
        tim_count,
        mvt_count,
        cavites_count,
        in_azi,
        catnat_inondation,
        zone_sismique,
        argiles_expo,
        radon_classe,
    ) = await asyncio.gather(
        georisques.count_basias(lat, lon, rayon_metres),
        georisques.count_ssp(lat, lon, rayon_metres),
        georisques.count_icpe(lat, lon, rayon_metres),
        georisques.count_tim(lat, lon, rayon_metres),
        georisques.count_mvt(lat, lon, rayon_metres),
        georisques.count_cavites(lat, lon, rayon_metres),
        georisques.in_azi(lat, lon, rayon_metres),
        georisques.catnat_inondation_count(code_insee) if code_insee else _none(),
        georisques.zonage_sismique(code_insee) if code_insee else _none(),
        georisques.argiles_exposition(code_insee) if code_insee else _none(),
        georisques.radon_classe(code_insee) if code_insee else _none(),
    )

    themes = [
        _theme_sols(basias_count, ssp_count),
        _theme_eau(in_azi, catnat_inondation),
        _theme_risques_naturels(mvt_count, cavites_count, zone_sismique, argiles_expo, radon_classe),
        _theme_activites_industrielles(icpe_count, tim_count),
    ]

    return SensitivityReport(
        address=address,
        rayon_metres=rayon_metres,
        themes=themes,
        niveau_global=worst_level([t.niveau for t in themes]),
    )


async def _none():
    return None


def _theme_sols(basias_count: int | None, ssp_count: int | None) -> ThemeSynthesis:
    niveau = worst_level([level_from_count(basias_count, seuil_moderee=1, seuil_elevee=4), level_from_ssp(ssp_count)])
    items: list[ThemeItem] = []
    manquantes: list[str] = []

    if basias_count is None:
        manquantes.append("anciens sites industriels (BASIAS)")
    else:
        items.append(
            ThemeItem(
                label="Anciens sites industriels ou d'activité de service",
                detail=f"{basias_count} site(s) recensé(s) à proximité" if basias_count else "Aucun site recensé à proximité",
                source="BASIAS — BRGM/Géorisques",
            )
        )
    if ssp_count is None:
        manquantes.append("sites et sols pollués (ex-BASOL)")
    else:
        items.append(
            ThemeItem(
                label="Sites et sols pollués, ou potentiellement pollués",
                detail=f"{ssp_count} site(s) recensé(s) à proximité" if ssp_count else "Aucun site recensé à proximité",
                source="SSP (ex-BASOL) — BRGM/Géorisques",
            )
        )

    resume = _resume_sols(niveau, basias_count, ssp_count)
    return ThemeSynthesis(key="sols", titre="Sols", niveau=niveau, resume=resume, items=items, donnees_manquantes=manquantes)


def _resume_sols(niveau, basias_count, ssp_count) -> str:
    if niveau == "indeterminee":
        return "Les bases de données sur les sols n'ont pas pu être interrogées."
    if ssp_count:
        return "Un ou plusieurs sites pollués ou potentiellement pollués sont recensés à proximité immédiate."
    if basias_count:
        return "Le secteur a accueilli une ou plusieurs activités industrielles ou de service par le passé, sans pollution confirmée à ce stade."
    return "Aucun site industriel ancien ni sol pollué n'est recensé à proximité dans les bases publiques."


def _theme_eau(in_azi: bool | None, catnat_inondation: int | None) -> ThemeSynthesis:
    niveau = level_from_flood_signals(in_azi, catnat_inondation)
    items: list[ThemeItem] = []
    manquantes: list[str] = []

    if in_azi is None:
        manquantes.append("zones inondables (AZI)")
    else:
        items.append(
            ThemeItem(
                label="Zone inondable recensée",
                detail="Le secteur recoupe une zone inondable connue" if in_azi else "Aucune zone inondable recensée à proximité",
                source="Atlas des zones inondables — Géorisques",
            )
        )
    if catnat_inondation is None:
        manquantes.append("arrêtés catastrophe naturelle (inondation)")
    else:
        items.append(
            ThemeItem(
                label="Historique catastrophe naturelle — inondation",
                detail=f"{catnat_inondation} arrêté(s) pris pour la commune" if catnat_inondation else "Aucun arrêté recensé pour la commune",
                source="GASPAR — Géorisques",
            )
        )

    if niveau == "indeterminee":
        resume = "Les indicateurs liés à l'eau n'ont pas pu être interrogés."
    elif niveau == "elevee":
        resume = "Le secteur est en zone inondable connue et la commune a déjà fait l'objet d'arrêtés catastrophe naturelle pour inondation."
    elif niveau == "moderee":
        resume = "Un signal lié au risque inondation existe (zone recensée ou antécédents communaux) : un avis hydrogéologique permettrait de préciser l'enjeu."
    else:
        resume = "Aucun signal notable lié au risque inondation n'est recensé à proximité dans les bases publiques consultées."

    return ThemeSynthesis(key="eau", titre="Eau", niveau=niveau, resume=resume, items=items, donnees_manquantes=manquantes)


def _theme_risques_naturels(
    mvt_count: int | None,
    cavites_count: int | None,
    zone_sismique: int | None,
    argiles_expo: str | None,
    radon_classe: int | None,
) -> ThemeSynthesis:
    niveaux = [
        level_from_count(mvt_count, seuil_moderee=1, seuil_elevee=3),
        level_from_count(cavites_count, seuil_moderee=1, seuil_elevee=3),
        level_from_zonage_sismique(zone_sismique),
        level_from_argiles(argiles_expo),
        level_from_radon(radon_classe),
    ]
    niveau = worst_level(niveaux)

    items: list[ThemeItem] = []
    manquantes: list[str] = []

    def add(value, label, detail_fn, source, missing_label):
        if value is None:
            manquantes.append(missing_label)
        else:
            items.append(ThemeItem(label=label, detail=detail_fn(value), source=source))

    add(mvt_count, "Mouvements de terrain recensés", lambda v: f"{v} évènement(s) recensé(s) à proximité" if v else "Aucun évènement recensé à proximité", "BRGM/Géorisques", "mouvements de terrain")
    add(cavites_count, "Cavités souterraines recensées", lambda v: f"{v} cavité(s) recensée(s) à proximité" if v else "Aucune cavité recensée à proximité", "BRGM/Géorisques", "cavités souterraines")
    add(zone_sismique, "Zonage sismique réglementaire", lambda v: f"Zone {v} sur l'échelle réglementaire (1 très faible à 5 fort)", "Géorisques", "zonage sismique")
    add(argiles_expo, "Retrait-gonflement des argiles", lambda v: f"Exposition {v.lower()}", "Géorisques", "retrait-gonflement des argiles")
    add(radon_classe, "Potentiel radon", lambda v: f"Classe {v} sur l'échelle réglementaire (1 à 3)", "Géorisques", "potentiel radon")

    if niveau == "indeterminee":
        resume = "Les indicateurs de risques naturels n'ont pas pu être interrogés."
    elif niveau == "elevee":
        resume = "Un ou plusieurs indicateurs de risques naturels (mouvements de terrain, sismicité, argiles ou radon) atteignent un niveau élevé sur le secteur."
    elif niveau == "moderee":
        resume = "Le secteur présente un ou plusieurs indicateurs de risques naturels à surveiller, sans signal alarmant à ce stade."
    else:
        resume = "Les indicateurs de risques naturels consultés sont globalement favorables sur le secteur."

    return ThemeSynthesis(
        key="risques_naturels", titre="Risques naturels", niveau=niveau, resume=resume, items=items, donnees_manquantes=manquantes
    )


def _theme_activites_industrielles(icpe_count: int | None, tim_count: int | None) -> ThemeSynthesis:
    niveau = worst_level(
        [
            level_from_count(icpe_count, seuil_moderee=1, seuil_elevee=3),
            level_from_count(tim_count, seuil_moderee=1, seuil_elevee=2),
        ]
    )
    items: list[ThemeItem] = []
    manquantes: list[str] = []

    if icpe_count is None:
        manquantes.append("installations classées (ICPE)")
    else:
        items.append(
            ThemeItem(
                label="Installations classées pour la protection de l'environnement",
                detail=f"{icpe_count} installation(s) recensée(s) à proximité" if icpe_count else "Aucune installation recensée à proximité",
                source="ICPE — Géorisques",
            )
        )
    if tim_count is None:
        manquantes.append("canalisations de transport de matières dangereuses")
    else:
        items.append(
            ThemeItem(
                label="Transport de matières dangereuses par canalisation",
                detail=f"{tim_count} canalisation(s) recensée(s) à proximité" if tim_count else "Aucune canalisation recensée à proximité",
                source="TIM — Géorisques",
            )
        )

    if niveau == "indeterminee":
        resume = "Les indicateurs d'activités industrielles n'ont pas pu être interrogés."
    elif niveau == "elevee":
        resume = "Plusieurs installations classées ou canalisations de matières dangereuses sont recensées à proximité immédiate."
    elif niveau == "moderee":
        resume = "Une ou plusieurs installations classées sont recensées à proximité, sans concentration particulière."
    else:
        resume = "Aucune installation classée ni canalisation à risque n'est recensée à proximité dans les bases publiques."

    return ThemeSynthesis(
        key="activites_industrielles", titre="Activités industrielles", niveau=niveau, resume=resume, items=items, donnees_manquantes=manquantes
    )
